### PR TITLE
Discuss triangular meshes in step-3.

### PR DESCRIPTION
**Pull request by bangerth**
In reference to #16944.

Something is not right. With the three changes mentioned (change mesh to triangles, use `FE_SimplexP`, and `QGaussSimplex`), the magnitude of the solution is 0.07266 whereas for the quad mesh it is 0.2949 (almost exactly four times as much). Anyone see what I'm doing wrong?

### CODE REVIEW

File: examples/step-3/doc/results.dox [hunk 1]
```diff
@@ -138,6 +138,63 @@ suggestions:
...
+  supports triangular and tetrahedral meshes. So a good experiment would
+  be to replace the mesh used here by a triangular mesh.
+
+  This is *almost* trivial. First, as discussed in step-1, we may want
+  to start with the quadrilateral mesh we are already creating, and
+  then convert it into a triangular one. You can do that by replacing
+  the first line of `Step3::make_grid()` by the following code:
+  @code
+  Triangulation<2> triangulation_quad;
+  GridGenerator::hyper_cube(triangulation_quad);
```
Comment (kronbichler) [inline examples/step-3/doc/results.dox:154]: I think you want to choose
```suggestion
  GridGenerator::hyper_cube(triangulation_quad, -1, 1);
```
This should then also fix the difference in the solution magnitude you observe.

File: examples/step-3/doc/results.dox [hunk 2]
```diff
@@ -138,6 +138,63 @@ suggestions:
...
+  GridGenerator::convert_hypercube_to_simplex_mesh (triangulation_quad,
+                                                    triangulation);
```
Comment (kronbichler) [inline examples/step-3/doc/results.dox:156]: I think you might want to mention that the this conversion function creates 8 triangles out of a single quadrilateral via https://github.com/dealii/dealii/blob/77ac9098c18b35709e63c6948f0da7d6b0b51258/source/grid/grid_generator.cc#L7932-L7943 so to reach the same number of global degrees of freedom, one needs to reduce the number of global refinements by 1. But I believe one needs more refinement to reach the same accuracy with simplices, so maybe 5 is not that wrong after all.

File: examples/step-3/doc/results.dox (hunk 1 above)
Comment (bangerth) [reply examples/step-3/doc/results.dox:154]: Oh, doh!

File: examples/step-3/doc/results.dox (hunk 2 above)
Comment (bangerth) [reply examples/step-3/doc/results.dox:156]: Yes, good idea. I mention it in passing, but don't dwell on it. That's for a later program.

**Comment by bangerth** [comment]
Updated as suggested.

File: examples/step-3/doc/results.dox (hunk 1 above)
Comment (bangerth) [reply examples/step-3/doc/results.dox]: That indeed fixes the problem, just for completeness.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**